### PR TITLE
Introduction of tito build system

### PR DESCRIPTION
**Pull request by mluscon**

### CODE REVIEW

File: VERSION.cmake [hunk 1]
```diff
@@ -1,3 +1,4 @@
-SET(HAWKEY_MAJOR "0")
-SET(HAWKEY_MINOR "5")
-SET(HAWKEY_PATCH "4")
+SET(HAWKEY_VERSION "0.5.3")
```
Comment (rholy) [inline VERSION.cmake]: The ```3``` is intentional? This is going to break CI since the current DNF requires hawkey >= 0.5.4

Comment (mluscon) [inline VERSION.cmake]: Switched back to 0.5.4

File: hawkey.spec [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
 Name:		hawkey
-Version:	@VERSION@
-Release:	1%{?dist}
+Version:	0.5.3
+Release:	2%{?dist}
 Summary:	Library providing simplified C and Python API to libsolv
 Group:		System Environment/Libraries
 License:	LGPLv2+
 URL:		https://github.com/rpm-software-management/%{name}
 # git clone https://github.com/rpm-software-management/hawkey.git && cd hawkey && package/archive
```
Comment (rholy) [inline hawkey.spec]: Please, update this comment too.

Comment (mluscon) [inline hawkey.spec]: Done

File: rel-eng/README [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Release instructions:
+
+1. Tag: tito tag
+2. Test: tito build --rpm --offline
```
Comment (rholy) [inline rel-eng/README]: Why ```--offline```?

**Comment by rholy** [comment]
```tito build --test --rpm``` still builds ```hawkey-0.5.3``` for me, so it is still going to break CI if I did not miss something.

File: hawkey.spec [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
-Release:	1%{?dist}
+Version:	0.5.3
+Release:	2%{?dist}
 Summary:	Library providing simplified C and Python API to libsolv
 Group:		System Environment/Libraries
 License:	LGPLv2+
 URL:		https://github.com/rpm-software-management/%{name}
-# git clone https://github.com/rpm-software-management/hawkey.git && cd hawkey && package/archive
-Source0:	https://github.com/rpm-software-management/%{name}/archive/%{gitrev}/%{name}-%{gitrev}.tar.gz
+# git clone https://github.com/rpm-software-management/hawkey.git && cd hawkey &&  tito build --tgz --test
```
Comment (rholy) [inline hawkey.spec]: ```tito build --tgz --test``` builds ```hawkey-git-0.%{version}.tar.gz``` on my system... it does not match ```%{name}-%{version}.tar.gz```

**Comment by rholy** [comment]
We already discussed it so I'm just posting it here to have it written somewhere: I don't agree with putting ```%{version}``` into ```Source0```

**Comment by cgwalters** [comment]
My 2¢: Using tito is better than hand-rolling yet another equivalent to it.  Other projects using it do put `%{version}` into `Source0`.  

See also https://fedoraproject.org/wiki/Layered_build_scripts_for_package_maintainers

**Comment by rholy** [comment]
Yes, but as I understand Fedora packaging guidelines, we should use hashes if we have the choice.

File: rel-eng/README (hunk 3 above)
Comment (mluscon) [inline rel-eng/README]: To prevent tito checking for nonexisting tag in remote repository

**Comment by mluscon** [comment]
> tito build --test --rpm still builds hawkey-0.5.3 for me, so it is still going to break CI if I did not miss something.

This is just a temporary issue that will be fixed by first tito tag commit.

**Pull request merged**